### PR TITLE
Phase 2 L1: add code for Phase 2 L1 GMT

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This is the 3rd PR to port all L1T Phase-2 developments to cms-sw. This PR targets the GMT code. It includes several PRs made locally, the details of which can be found in the twiki https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideL1TPhase2Instructions and in the cms-l1t-offline branch.

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
@perrotta, @dpiparo, @qliphy Could you please help me understand what needs to be changed to fix these header consistency issues? How can I test this locally without restarting the tests?

**Comment by perrotta** [comment]
> @perrotta, @dpiparo, @qliphy Could you please help me understand what needs to be changed to fix these header consistency issues? How can I test this locally without restarting the tests?

Looking at the error log [1], it looks like L1Trigger/Phase2L1GMT/interface/TrackMuonMatchAlgorithm.h misses header includes of the classes mentioned therein
I would go through the other issues mentioned in that log to see if there is anything else unrelated

[1] https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-762f05/23440/headers_chks.log

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/Phase2L1GMT/src/L1TPhase2GMTEndcapStubProcessor.cc [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+  int wheel = 0;
+  int sign = endcap == 1 ? -1 : 1;
+
+  if (ring == 3)
+    wheel = sign * 3;
+
+  if (ring == 2)
+    wheel = sign * 4;
+  if (ring == 1)
+    wheel = sign * 5;
```
Comment (perrotta) [inline L1Trigger/Phase2L1GMT/src/L1TPhase2GMTEndcapStubProcessor.cc:47]:
```suggestion
  if (ring == 3)
    wheel = sign * 3;
  else if (ring == 2)
    wheel = sign * 4;
  else if (ring == 1)
    wheel = sign * 5;
```

File: L1Trigger/Phase2L1GMT/interface/Isolation.h [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      }
+    }
+    nevto++;
+  }
+
+  inline int Isolation::SetAbsIsolationBits(int accum) {
+    int iso = (accum <= absiso_thrT ? 3 : accum <= absiso_thrM ? 2 : accum <= absiso_thrL ? 1 : 0);
+
+    if (verbose_) {
+      cout << " [DEBUG Isolation] : absiso_threshold L : " << absiso_thrL << " accum " << accum
```
Comment (qliphy) [inline L1Trigger/Phase2L1GMT/interface/Isolation.h:103]: Here and other places:  std::cout is not threadsafe, please use https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Pull request merged**